src/lambda_handler.py

```diff
@@ -109,19 +229,75 @@ def handler(event, context) -> dict[str, Optional[str]]:
         old_log_level = logging.getLogger().getEffectiveLevel()
         logging.getLogger().setLevel(new_log_level)
 
-    task_name = f"task_{event.get('task')}"
-    task = globals().get(task_name, task_default)
+    # Set up the SSM client if necessary
+    if ssm_client is None:
+        ssm_client = boto3_client("ssm")
+
+    mongodb_uri_elements: List[Tuple[str, Optional[str]]] = []
+
+    # This only runs from a CloudWatch scheduled event invocation
+    trigger_source: Optional[str]
+    trigger_type: Optional[str]
+    if (
+        (trigger_source := event.get("source")) is None
+        or trigger_source != "aws.events"
+    ) or (
+        (trigger_type := event.get("detail-type")) is None
+        or trigger_type != "Scheduled Event"
+    ):
+        logging.error("Invalid invocation event.")
+        return
+
+    for var in [
+        "ssm_db_user",
+        "ssm_db_pass",
+        "ssm_db_host",
+        "ssm_db_port",
+        "ssm_db_authdb",
+    ]:
+        mongodb_uri_elements.append((var, os.environ.get(var)))
 
-    result: dict[str, Any]
-    if not callable(task):
-        logging.error("Provided task is not a callable.")
-        logging.error(task)
-        result = task_default(event)
+    # Check that we have all of the required variables
+    if missing_variables := [k for k, v in mongodb_uri_elements if v is None]:
+        logging.error("Missing required variables: %s", ",".join(missing_variables))
+        return
+
+    # Determine the database where the CVE data will be inserted
+    write_db = get_ssm_parameter(os.environ.get("ssm_db_writedb", "ssm_db_authdb"))
+
+    # Determine if a non-default CVEs JSON URL is being used
+    nvd_urls = generate_urls()
+
+    json_url = os.environ.get("json_url")
+    if json_url is None:
+        cve_json_urls = nvd_urls
     else:
-        result = task(event)
-    response.update(result)
+        cve_json_urls = [json_url]
+
+    # Determine if a non-default collection is being used
+    db_collection = os.environ.get("target_collection")
+    if db_collection is not None:
+        CVEDoc.Settings.name = get_ssm_parameter(db_collection)
+
+    # We disable mypy here because the variable is typed to have Optional[str] elements
+    # but we verify that there are only str elements before this point.
+    mongodb_uri = build_mongodb_uri(*[v for k, v in mongodb_uri_elements])  # type: ignore
+
+    if mongodb_uri is None:
+        logging.error("Unable to import KEV data.")
+        return
+
+    # Set up the Motor session if necessary
+    if motor_client is None:
+        motor_client = AsyncIOMotorClient(mongodb_uri)
+
+    try:
+        asyncio.get_event_loop().run_until_complete(
+            process_urls(cve_json_urls, write_db)
+        )
```

Comment:
Would it be simpler to use a `TaskGroup` here?

Reply:
`async.gather()` is the old way, from before `TaskGroup` became a thing.  You can just use that.

Reply:
Task groups were I believe introduced in [Python 3.11](https://www.python.org/downloads/release/python-3110/) no? Until the way we do Lambdas is updated (work in progress) the repos are still designed to support multiple Python versions.

Reply:
I am not sure if we can put a `TaskGroup` or rather `async.gather()`. We would need to make the lambda `handler` function async and when I attempted to do that I kept running into errors. I imagine it is possible to get it to run asynchronously but it wasn't a simple change without looking under the hood of how we are running the lambda handler.

Reply:
Can you just add a comment that this is icky and we should revisit it to find a better solution?

Reply:
@jsf9k I addressed this here: https://github.com/cisagov/cyhy-nvdsync-lambda/pull/2/commits/e6397c680698de8835b0629e200b28fe2829df5b#:~:text=%23%20This%20could%20be,an%20asynchronous%20environment.